### PR TITLE
Remove scala.meta.{Symbol,Signature}

### LINKED ISSUE #1525: Remove scala.meta.{Symbol,Signature}
Now that SemanticDB symbols are primarily strings, we should no longer expose `scala.meta.{Symbol,Signature}` in the public API.

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes #1525. This PR replaces usage of `scala.meta.{Symbol,Signature}` with `String`. The refactoring was straightforward in most cases except for `Symbol.Multi`. Multi symbols are not in the spec since they are not persisted and this PR does not change that.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -364,18 +364,6 @@ lazy val treesJVM = trees.jvm
...
-  .in(file("scalameta/semanticdb"))
-  .settings(
-    publishableSettings,
-    description := "Scalameta semantic database APIs"
-  )
-  .nativeSettings(nativeSettings)
-  .dependsOn(common)
-lazy val semanticdbJVM = semanticdb.jvm
-lazy val semanticdbJS = semanticdb.js
-lazy val semanticdbNative = semanticdb.native
```
Comment (xeno-by) [inline build.sbt]: 😍 I will proceed with https://github.com/scalameta/scalameta/issues/1595 when I return from ScalaDays.

File: semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala [hunk 2]
```diff
@@ -8,34 +8,35 @@ import scala.meta.internal.{semanticdb3 => s}
 import scala.meta.internal.semanticdb3.Scala._
 import scala.meta.internal.semanticdb3.Scala.{Descriptor => d}
 import scala.util.control.NonFatal
+import scala.meta.internal.semanticdb3.Scala._
```
Comment (xeno-by) [inline semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala]: There is already the same import on line 8.

File: semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala [hunk 3]
```diff
@@ -8,34 +8,35 @@ import scala.meta.internal.{semanticdb3 => s}
...
+        if (sym.isOverloaded) return Symbols.Multi(sym.alternatives.map(_.toSemantic))
         if (sym.isModuleClass) return sym.asClass.module.toSemantic
         if (sym.isTypeSkolem) return sym.deSkolemize.toSemantic
         if (sym.isSemanticdbLocal) return freshSymbol(sym.pos.toMeta.input)
 
         val owner = sym.owner.toSemantic
         val signature = {
           if (sym.isMethod || sym.isUsefulField) {
-            m.Signature.Method(sym.name.toSemantic, sym.disambiguator)
+            Descriptor.Method(sym.name.toSemantic, sym.disambiguator)
```
Comment (xeno-by) [inline semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala]: The convention that I've been used in other parts of Scalameta is to say `d.Method`, `d.TypeParameter`, etc. There is already `import scala.meta.internal.semanticdb3.Scala.{Descriptor => d}` in scope in this file.

File: scalameta/semanticdb/shared/src/main/scala/scala/meta/semanticdb/Symbol.scala [hunk 4]
```diff
@@ -1,203 +0,0 @@
...
-
-  final case class Local(id: String) extends Symbol {
-    override def toString = syntax
-    override def syntax = Symbols.Local(id)
-    override def structure = s"""Symbol.Local("$id")"""
-  }
-
-  final case class Global(owner: Symbol, signature: Signature) extends Symbol {
-    override def toString = syntax
-    override def syntax = Symbols.Global(owner.syntax, signature.syntax)
```
Comment (xeno-by) [inline scalameta/semanticdb/shared/src/main/scala/scala/meta/semanticdb/Symbol.scala]: Now that this line is gone, I suggest we delete `def Global(owner: String, desc: String): String = ???` from the public API.

File: scalameta/semanticdb/shared/src/main/scala/scala/meta/semanticdb/Symbol.scala [hunk 5]
```diff
@@ -1,203 +0,0 @@
...
-object Symbol {
-  case object None extends Symbol {
-    override def toString = syntax
-    override def syntax = Symbols.None
-    override def structure = s"""Symbol.None"""
-  }
-
-  final case class Local(id: String) extends Symbol {
-    override def toString = syntax
-    override def syntax = Symbols.Local(id)
```
Comment (xeno-by) [inline scalameta/semanticdb/shared/src/main/scala/scala/meta/semanticdb/Symbol.scala]: Now that this line is gone, I suggest we delete `def Local(id: String): String = ???` from the public API.

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala [hunk 6]
```diff
@@ -14,6 +13,8 @@ object Scala {
     def Global(owner: String, desc: String) = if (owner != RootPackage) owner + desc else desc
     def Local(id: Int): String = Local("local" + id.toString)
     def Local(symbol: String): String = symbol
+    def Multi(symbols: List[String]): String = symbols.mkString(";", ";", "")
+    def parse(symbol: String): (String, Descriptor) = DescriptorParser(symbol).swap
```
Comment (xeno-by) [inline semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: Looks like this method is only used in tests. If that's the intent, let's move the method into tests.

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala [hunk 7]
```diff
@@ -22,6 +23,31 @@ object Scala {
     def isEmptyPackage: Boolean = symbol == Symbols.EmptyPackage
     def isGlobal: Boolean = !isNone && Descriptor.descriptorLasts.contains(symbol.last)
     def isLocal: Boolean = !isNone && !isGlobal
+    def isMulti: Boolean = symbol.startsWith(";")
+    def flattenMulti: List[String] = {
```
Comment (xeno-by) [inline semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: How about we rename this to `symbols` or `parts`?

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala [hunk 8]
```diff
@@ -22,6 +23,31 @@ object Scala {
     def isEmptyPackage: Boolean = symbol == Symbols.EmptyPackage
     def isGlobal: Boolean = !isNone && Descriptor.descriptorLasts.contains(symbol.last)
```
Comment (xeno-by) [inline semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: This should be updated to take multi symbols into account.

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala [hunk 9]
```diff
@@ -22,6 +23,31 @@ object Scala {
...
     def isLocal: Boolean = !isNone && !isGlobal
```
Comment (xeno-by) [inline semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: Ditto.

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala (hunk 6 above)
Comment (olafurpg) [reply semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: `DescriptorParser.apply` is private, can I make it `private[meta]` instead?

Comment (xeno-by) [reply semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: Sure.

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala (hunk 7 above)
Comment (olafurpg) [reply semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: I agree this name is not ideal, it replaced a `def flatten(sym: Symbol): Symbol` in metac but `String#flatten` is already used by the scala-library. I would like to have "multi" somewhere in there, how about `asMulti: List[String]`?

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala (hunk 8 above)
Comment (olafurpg) [reply semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: You mean by adding `!isMulti && ...`? I don't think we should do `asMulti.exists(_.isGlobal)`

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala (hunk 7 above)
Comment (xeno-by) [reply semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: Sounds good.

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala (hunk 8 above)
Comment (olafurpg) [reply semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: Done, made it
```scala
    def isGlobal: Boolean =
      !isNone &&
        Descriptor.descriptorLasts.contains(symbol.last) &&
        (!isMulti || asMulti.exists(_.isGlobal))
    def isLocal: Boolean =
      !isNone &&
        !isGlobal &&
        (!isMulti || asMulti.exists(_.isLocal))
```

File: scalameta/semanticdb/shared/src/main/scala/scala/meta/semanticdb/Symbol.scala (hunk 4 above)
Comment (olafurpg) [reply scalameta/semanticdb/shared/src/main/scala/scala/meta/semanticdb/Symbol.scala]: Done.

File: semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala (hunk 2 above)
Comment (olafurpg) [reply semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala]: Removed, good catch!

File: semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala (hunk 3 above)
Comment (olafurpg) [reply semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolOps.scala]: Fixed.

File: semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala (hunk 6 above)
Comment (olafurpg) [reply semanticdb/semanticdb3/src/main/scala/scala/meta/internal/semanticdb3/Scala.scala]: Done.

**Pull request merged**